packages/core/src/delegation.js

```diff
@@ -288,6 +296,69 @@ export class Delegation {
   }
 }
 
+/**
+ * Writes given `Delegation` chain into a content addressed archive (CAR)
+ * buffer and returns it.
+ *
+ * @param {API.Delegation} delegation}
+ */
+export const archive = async delegation => {
+  try {
+    // Iterate over all of the blocks in the DAG and add them to the
+    // block store.
+    const store = new Map()
+    for (const block of delegation.iterateIPLDBlocks()) {
+      store.set(`${block.cid}`, block)
+    }
+
+    // Then we we create a descriptor block to describe what this DAG represents
+    // and it to the block store as well.
+    const variant = await CBOR.write({
+      [`ucan@${delegation.version}`]: delegation.root.cid,
+    })
+    store.set(`${variant.cid}`, variant)
+
+    // And finally we encode the whole thing into a CAR.
+    const bytes = CAR.encode({
+      roots: [variant],
+      blocks: store,
+    })
+
+    return ok(bytes)
+  } catch (cause) {
+    return error(/** @type {Error} */ (cause))
+  }
+}
+
+export const ArchiveSchema = Schema.variant({
+  'ucan@0.9.1': /** @type {Schema.Schema<API.UCANLink>} */ (
```

Comment:
Should this version number be a constant somewhere?

Reply:
It is the `UCAN.VERSION`, however at some point we'll probably upgrade it to next version, yet version here should remain the same. I'm not sure there is a value in defining a new constant somewhere if this is the only thing referring to it though.